packages/nodes-base/nodes/ExecuteWorkflow/GenericFunctions.ts

```diff
@@ -29,7 +29,8 @@ export async function getWorkflowInfo(this: IExecuteFunctions, source: string, i
 		workflowInfo.code = jsonParse(workflowJson);
 	} else if (source === 'parameter') {
 		// Read workflow from parameter
-		workflowInfo.code = this.getNodeParameter('workflowJson', itemIndex) as IWorkflowBase;
+		const workflowJson = this.getNodeParameter('workflowJson', itemIndex) as string;
```

Comment:
A better fix for this would be to make sure that `getNodeParameter` does `jsonParse` when the parameter type is `json`. either that, or we get rid of `type: 'json'`, and convert such parameters to be strings.